multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutesValidator.java

```diff
@@ -74,22 +73,42 @@ public Object attemptToCorrect(Object routes, final Map<String, Object> context)
         }
 
         return routesList.stream()
-                         .map(route -> attemptToCorrectParameterMap(route, context))
+                         .map(routesElement -> getRelatedParametersForContext(routesElement, context))
+                         .map(this::attemptToCorrectParameterMap)
                          .collect(Collectors.toList());
     }
 
-    private Map<String, Object> attemptToCorrectParameterMap(Map<String, Object> originalElem, Map<String, Object> context) {
+    private Map<String, Object> getRelatedParametersForContext(Map<String, Object> routesElement, Map<String, Object> context) {
+        Map<String, Object> updatedContext = new HashMap<>(routesElement);
+        Set<String> relatedParameters = getRelatedParameterNames();
+        Map<String, Object> updatedParamters = relatedParameters.stream()
+                                                                .filter(relatedParameter -> doesParameterMissingInRoutesElement(relatedParameter,
+                                                                                                                                routesElement,
+                                                                                                                                context))
+                                                                .collect(Collectors.toMap(Function.identity(), context::get));
+        updatedContext.putAll(updatedParamters);
+        return updatedContext;
+    }
+
+    private boolean doesParameterMissingInRoutesElement(String relatedParameter, Map<String, Object> routesElement,
```

Comment:
rename this method to isParameterMissingInRoutes or maybe even isGlobalParameterValueOverridden